app/classes/mappable/collapsible_collection_of_objects.rb

```diff
@@ -106,28 +106,43 @@ def round_number(num, prec)
       (num.to_f * prec).round.to_f / prec
     end
 
-    # Sets with unknown location are messing up the maps.
-    # Dismiss any set that has an unknown location.
+    # Dismiss any set that has an unknown location. Also dismiss vague locations
+    # e.g. "Nova Scotia" in collections. (ok for single locations, sets of 1)
+    # Mapsets containing unknown or vague locations will obscure useful data
+    # if GPS points or useful locations are grouped with huge nonspecific areas.
+    def mappable_location?(is_collection, obj)
+      return true unless is_collection
+
+      if obj.location?
+        !obj.vague? && !Location.is_unknown?(obj.name)
+      elsif obj.observation?
+        !obj.location&.vague? && !Location.is_unknown?(obj.location&.name)
+      end
+    end
```

Comment:
If obj can only be a Location or an Observation, then 
- consider `else` instead of `elsif` 
- and/or consider adding an `else` that does some error handling or just returns `false`.

Reply:
Revised